docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+The key provider service program encrypts the `PrivateLayerBlockCipherOptions` with the user's OpenPGP key. After that, it will perform Base64 encoding and then package it into a JSON structure. Under the confidential container system, it is called 'annotation packet' to record some additional information, such as the ID of the user's private key, for example:
+
+```
+{
+    "key_id":"1234",
+    "wrapped_data":#base64encode(Enc(PrivateLayerBlockCipherOptions)),
+    "wrap_type":"",
+}
+```
+
+The specific format of the annotation packet can be determined by the key provider program used by the user, but it is necessary to ensure that the key provider program used by the decryptor (i.e. the attestation-agent integrated with the specified Key Broker Client) supports parsing annotation packets of the same format.
+
+After completing the above work, Base64 encode the `PublicLayerBlockCipherOptions` and annotation packet, and write them into the manifest.json of the container image as the annotation field of the layer, the former is placed in the `org.opencontainers.image.enc.pubopts` field and the latter in the `org.opencontainers.image.enc.keys.provider.attestation-agent` field. Finally, add the "+encrypted" suffix to the mediaType field of this layer. For example, the container image manifest with an encrypted layer is as follows:
+
+```json
+{
+    "schemaVersion": 2,
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:ef87d9f0...",
+        "size": 1510
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
+            "digest": "sha256:f69ae40...",
+            "size": 204812,
+            "annotations": {
+                "org.opencontainers.image.enc.pubopts": "eyJjaXBoZXIiOi...",
+                "org.opencontainers.image.enc.keys.provider.attestation-agent": "eyJwcm9..."
+            }
+        }
+    ]
+}
+```
+
+### Image signing
+
+The container image signing standard used by the confidential containers V1 is follow the signing mechanism used in [containers/image](https://github.com/containers/image). At present, the only defined cryptographic signature format is a OpenPGP signature (RFC 4880). The signing process will not modify any content of the container image, including layer, config, manifest, etc. Therefore, the signature will not change the image manifest digest used to identify the image.
+
+The container signature under this standard is essentially a blob, which consists of two parts:
+
+- JSON payload
+
+- Cryptographic signature of JSON payload
+
+The JSON payload contains the manifest digest and identity information of the container image. Examples are as follows:
+
+```json
+{
+    "critical": {
+        "type": "atomic container signature",
+        "image": {
+            "docker-manifest-digest": "sha256:817a12c32a39be..."
```

Comment:
Why "docker"? Why not just "manifest-digest"? Or "unencrypted-manifest-digest" if that is the intent?

Reply:
As above.